artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -530,4 +543,17 @@ private static byte destinationType(Destination destination) {
       throw new IllegalArgumentException("Unknown Destination Type passed to JMS Transformer.");
    }
 
+   private static byte destinationType(SimpleString destination) {
+      if (destination.startsWith(ActiveMQDestination.QUEUE_QUALIFIED_PREFIX_SIMPLE))
+         return QUEUE_TYPE;
+      if (destination.startsWith(ActiveMQDestination.TEMP_QUEUE_QUALIFED_PREFIX_SIMPLE))
+         return TEMP_QUEUE_TYPE;
+      if (destination.startsWith(ActiveMQDestination.TOPIC_QUALIFIED_PREFIX_SIMPLE))
+         return TOPIC_TYPE;
+      if (destination.startsWith(ActiveMQDestination.TEMP_TOPIC_QUALIFED_PREFIX_SIMPLE))
+         return TEMP_TOPIC_TYPE;
+
+      throw new IllegalArgumentException("Unknown Destination Type passed to JMS Transformer.");
```

Comment:
This will break stuff not everyone does jms. It’s valid to send without a jms type.

Reply:
It's only used for JMSReplyTo property. but I will return Queue (as it does now) just in case.